src/sdk/pynni/nni/bohb_advisor/bohb_advisor.py

```diff
@@ -0,0 +1,594 @@
+        self.credit = 0
+        self.brackets = dict()
+        self.search_space = None
+        # [key, value] = [parameter_id, parameter]
+        self.parameters = dict()
+
+        # config generator
+        self.cg = None
+
+    def load_checkpoint(self):
+        pass
+
+    def save_checkpoint(self):
+        pass
+
+    def handle_initialize(self, data):
+        """Initialize Tuner, including creating Bayesian optimization-based parametric models 
+        and search space formations
+
+        Parameters
+        ----------
+        data: search space
+            search space of this experiment
+        """
+        logger.info('start to handle_initialize')
+        # convert search space jason to ConfigSpace
+        self.handle_update_search_space(data)
+
+        # generate BOHB config_generator using Bayesian optimization
+        if not self.search_space is None:
+            self.cg = CG_BOHB(configspace=self.search_space,
+                              min_points_in_model=self.min_points_in_model,
+                              top_n_percent=self.top_n_percent,
+                              num_samples=self.num_samples,
+                              random_fraction=self.random_fraction,
+                              bandwidth_factor=self.bandwidth_factor,
+                              min_bandwidth=self.min_bandwidth)
+        else:
+            raise ValueError('Error: Search space is None')
+        # generate first brackets
+        self.generate_new_bracket()
+        send(CommandType.Initialized, '')
+        return True
+
+    def generate_new_bracket(self):
+        """generate a new bracket"""
+        logger.debug(
+            'start to create a new SuccessiveHalving iteration, self.curr_s=%d', self.curr_s)
+        if self.curr_s < 0:
+            logger.info("s < 0, Finish this round of Hyperband in BOHB. Generate new round")
+            self.curr_s = self.s_max
+        self.brackets[self.curr_s] = Bracket(s=self.curr_s, s_max=self.s_max, eta=self.eta,
+                                    max_budget=self.max_budget, optimize_mode=self.optimize_mode)
+        next_n, next_r = self.brackets[self.curr_s].get_n_r()
+        logger.debug(
+            'new SuccessiveHalving iteration, next_n=%d, next_r=%d', next_n, next_r)
+        # rewrite with TPE
+        generated_hyper_configs = self.brackets[self.curr_s].get_hyperparameter_configurations(
+            next_n, next_r, self.cg)
+        self.generated_hyper_configs = generated_hyper_configs.copy()
+
+    def handle_request_trial_jobs(self, data):
+        """recerive the number of request and generate trials
+
+        Parameters
+        ----------
+        data: int
+            number of trial jobs that nni manager ask to generate
+        """
+        # Receive new request
+        self.credit += data
+
+        for _ in range(self.credit):
+            self._request_one_trial_job()
+
+        return True
```

Comment:
always ready True?（seems meaningless to return?） Seems no docstring to explain the return type?

Reply:
return True to indicate that the request has been received. Add docstring already.

Reply:
Done.